### PR TITLE
Add initial beatmap & beatmapset database schema

### PR DESCRIPTION
**Pull request by Lekuruu**
Here is a schema for beatmaps I came up with, to get things moving forward a bit. Let me know if I missed anything!

### CODE REVIEW

File: migrations/002_Beatmaps.up.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    approved_by integer NOT NULL REFERENCES users (id),
+    status smallint NOT NULL DEFAULT -3,
+    description text NOT NULL DEFAULT '',
+    has_video boolean NOT NULL DEFAULT false,
+    availability_status smallint NOT NULL DEFAULT 0,
+    availability_info text NOT NULL DEFAULT ''
+);
+
+CREATE TABLE beatmaps (
+    id serial NOT NULL PRIMARY KEY REFERENCES beatmapsets (id),
```
Comment (ascenttree) [inline migrations/002_Beatmaps.up.sql]: Shouldn't the reference be on `set_id`?
```suggestion
    id serial NOT NULL PRIMARY KEY,
    set_id integer NOT NULL REFERENCES beatmapsets (id),
```

File: migrations/002_Beatmaps.up.sql [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    source character varying(255) NOT NULL,
+    creator_id integer NOT NULL REFERENCES users (id),
+    created_at timestamp without time zone NOT NULL DEFAULT now(),
+    last_updated timestamp without time zone NOT NULL DEFAULT now(),
+    approved_at timestamp without time zone NOT NULL DEFAULT now(),
+    approved_by integer NOT NULL REFERENCES users (id),
+    status smallint NOT NULL DEFAULT -3,
+    description text NOT NULL DEFAULT '',
+    has_video boolean NOT NULL DEFAULT false,
+    availability_status smallint NOT NULL DEFAULT 0,
```
Comment (ascenttree) [inline migrations/002_Beatmaps.up.sql:14]: What's availability?

**Review by ascenttree — APPROVED** [review]

**Comment by ascenttree** [comment]
I did not mean to approve that
Oops

**Comment by ascenttree** [comment]
What have  idone

File: migrations/002_Beatmaps.up.sql (hunk 2 above)
Comment (Lekuruu) [reply migrations/002_Beatmaps.up.sql:14]: DMCA Stuff

File: migrations/002_Beatmaps.up.sql (hunk 1 above)
Comment (Lekuruu) [reply migrations/002_Beatmaps.up.sql]: No, this targets the beatmapsets table

Comment (ascenttree) [reply migrations/002_Beatmaps.up.sql]: ?, why does the beatmap (meaning diff?) id relate to the beatmapset instead of the set id

Comment (Lekuruu) [reply migrations/002_Beatmaps.up.sql]: The beatmapset table does not have a `set_id` column. It uses an `id` column.

Comment (ascenttree) [reply migrations/002_Beatmaps.up.sql]: The beatmaps table does have a set_id column though, that relates to nothing

Comment (Lekuruu) [reply migrations/002_Beatmaps.up.sql:19]: Even though I just woke up, I can say that the beatmapsets table does not have that

Comment (Lekuruu) [reply migrations/002_Beatmaps.up.sql]: Nevermind, I should never do reviews at 5 am

File: migrations/002_Beatmaps.up.sql [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    availability_info text NOT NULL DEFAULT ''
+);
+
+CREATE TABLE beatmaps (
+    id serial NOT NULL PRIMARY KEY,
+    set_id integer NOT NULL REFERENCES beatmapsets (id),
+    set_id integer NOT NULL,
```
Comment (Lekuruu) [inline migrations/002_Beatmaps.up.sql]: @ascenttree I think you forgot this line

**Review by ascenttree — APPROVED** [review]

**Comment by Lekuruu** [comment]
I'll re-test once I am home

**Comment by ascenttree** [comment]
Just tested this, seems to work fine for me. I guess now is beatmap submission :despairge:

**Pull request merged**